fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreIndexTest.java

```diff
@@ -2358,6 +2366,8 @@ public void boundaryPrimaryKeys() throws Exception {
         checkSplitIndexBuildRange(boundaryPrimaryKeysSize / 2, Integer.MAX_VALUE, oneRangePerSplit, indexer); // to test that integer overflow isn't a problem
 
         indexer.close();
+        factory.unsetLocalityProvider();
```

Comment:
This will still have problems if the test fails -- you'll never reset the provider.  How about something like:

    @Test   
    public void testSomething() {
        runLocalityTest( () -> testSomethingImpl() );
    }

    private void testSomethingImpl() {
    ...
    }

    private void runLocalityTest(Runnable test) {
        final FDBLocalityProvider origProvider = factory.getLocalityProvider();
        try {
            test.run();
        } finally {
            factory.setLocalityProvider(origProvider);
        }
    }

Reply:
Yeah you are right. Did what you suggested.